src/radical/pilot/agent/staging_input/default.py

```diff
@@ -199,7 +200,13 @@ def _handle_unit(self, unit, actionables):
                     self._log.error('no transfer for %s -> %s', src, tgt)
                     self._prof.prof('staging_in_fail', uid=uid, msg=did)
                     raise NotImplementedError('unsupported transfer %s' % src)
-
+            elif action == rpc.TARBALL:
+                # If somethig was staged via the tarball method, the tarball is
+                # extracted and then removed from the unit folder.
+                tar = tarfile.open(tgtdir+'/'+uid+'.tar')
+                tar.extractall(path=tgtdir)
+                tar.close()
+                os.remove(tgtdir+'/'+uid+'.tar')
```

Comment:
You may want to keep the tarballs if `RADICAL_DEBUG` is set?

Reply:
Are the files the user transferred enough? I agree it needs a debug message to show what is happening.